### PR TITLE
Get class data

### PR DESCRIPTION
**Pull request by Alexmhaley**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -11,3 +11,4 @@ SENTRY_ENVIRONMENT=dev
 ELASTICSEARCH_HOSTS=http://localhost:9200
 AUTH_SERVICE_BASE_URL=http://localhost:5001
 SERVICE_JWT=
+SCHOOL_USER_ASSOC_SERVICE_BASE_URL=http://localhost:5027
```
Comment (thomas5631) [inline .env.example]: Should you add the class roster base url here too?

File: charts/user-search-worker/configuration/ci.yaml [hunk 2]
```diff
@@ -19,3 +19,5 @@ SENTRY_DSN: 'https://4cfd1fd22d0746c08d181afa9b1c8ef1@sentry.io/1763582'
 SENTRY_ENABLED: 'true'
 SENTRY_ENVIRONMENT: 'ci'
 SERVICE_JWT: '__ssm:SERVICE_JWT'
+SCHOOL_USER_ASSOC_SERVICE_BASE_URL: 'http://qa-ts-gateway-svc-gateway-service.qa.svc.cluster.local/svc/school-user-assoc'
+CLASS_ROSTER_SERVICE_BASE_URL: 'http://qa-tsc-class-roster-svc-class-roster-service.qa.svc.cluster.local:8080'
```
Comment (thomas5631) [inline charts/user-search-worker/configuration/ci.yaml]: would you be able to order these alphabetically?

File: src/handlers/syncUsers/dataProcessors/getAuthData.ts [hunk 3]
```diff
@@ -1,26 +1,102 @@
...
+async function addSchoolData(
+    APIs: { schoolUserAssocService: SchoolUserAssocServiceAPI },
+    users,
+): Promise<SchoolAssocServiceUser[]> {
+    const userDetails = await getSchoolAssocData(APIs, users);
+    return userDetails;
+}
+
+async function addClassData(APIs: { classRosterService: ClassRosterServiceAPI }, users): Promise<User[]> {
+    console.log('***************');
```
Comment (thomas5631) [inline src/handlers/syncUsers/dataProcessors/getAuthData.ts]: console.log

File: src/handlers/syncUsers/dataProcessors/getAuthData.ts [hunk 4]
```diff
@@ -1,26 +1,102 @@
...
+    const userDetails = await getClassRosterData(APIs, users);
+    console.log(userDetails);
```
Comment (thomas5631) [inline src/handlers/syncUsers/dataProcessors/getAuthData.ts]: console.log

File: src/handlers/syncUsers/dataProcessors/getClassRosterData.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            }),
+        );
+    } catch (err) {
+        throw Error(err);
+    }
+    const outputData: User[] = [];
+    userDetails.forEach((element, index) => {
+        const newObj: User = { ...element, classes: [] };
+        newObj.classes = classDetails[index].classes;
+        console.log(newObj.classes);
```
Comment (thomas5631) [inline src/handlers/syncUsers/dataProcessors/getClassRosterData.ts]: console.log

File: src/types/interfaces.d.ts [hunk 6]
```diff
@@ -18,3 +18,83 @@ interface Beeline {
...
+}
+
+interface Class {
+    classRosterId: string;
+    name: string;
+    nickname: string;
+    isMain: boolean;
+}
+
+interface ClassData {
```
Comment (thomas5631) [inline src/types/interfaces.d.ts]: Would this be CollatedClassUser?

File: src/types/interfaces.d.ts [hunk 7]
```diff
@@ -76,3 +92,21 @@ interface CollatedSchoolUser {
...
+
+interface APIClass {
+    classRosterId: string;
+    name: string;
+    nickname: string;
+    isMain: boolean;
+}
+interface Class {
+    class_roster_id: string;
+    name: string;
```
Comment (thomas5631) [inline src/types/interfaces.d.ts]: Should this be class_name?

File: src/types/interfaces.d.ts [hunk 8]
```diff
@@ -76,3 +92,21 @@ interface CollatedSchoolUser {
...
 }
+
+interface APIClass {
+    classRosterId: string;
+    name: string;
+    nickname: string;
+    isMain: boolean;
+}
+interface Class {
+    class_roster_id: string;
```
Comment (thomas5631) [inline src/types/interfaces.d.ts]: Should this be class_id?

File: src/types/interfaces.d.ts [hunk 9]
```diff
@@ -76,3 +92,21 @@ interface CollatedSchoolUser {
...
+    name: string;
+    nickname: string;
```
Comment (thomas5631) [inline src/types/interfaces.d.ts]: Should this be class_nickname?

File: charts/user-search-worker/configuration/ci.yaml [hunk 10]
```diff
@@ -20,3 +21,4 @@ SENTRY_DSN: 'https://4cfd1fd22d0746c08d181afa9b1c8ef1@sentry.io/1763582'
 SENTRY_ENABLED: 'true'
 SENTRY_ENVIRONMENT: 'ci'
 SERVICE_JWT: '__ssm:SERVICE_JWT'
+SCHOOL_USER_ASSOC_SERVICE_BASE_URL: 'http://qa-ts-gateway-svc-gateway-service.qa.svc.cluster.local/svc/school-user-assoc'
```
Comment (thomas5631) [inline charts/user-search-worker/configuration/ci.yaml]: This has already been added above

File: src/types/interfaces.d.ts (hunk 9 above)
Comment (Alexmhaley) [reply src/types/interfaces.d.ts]: kind of assumed that as it's on a class object already we didn't need to be too specific but if you'd rather go more specific I'm happy to change it

File: charts/user-search-worker/configuration/ci.yaml (hunk 2 above)
Comment (Alexmhaley) [reply charts/user-search-worker/configuration/ci.yaml]: Yep have re-ordered correctly

File: src/handlers/syncUsers/dataProcessors/getAuthData.ts (hunk 3 above)
Comment (Alexmhaley) [reply src/handlers/syncUsers/dataProcessors/getAuthData.ts]: Removed

File: src/handlers/syncUsers/dataProcessors/getAuthData.ts (hunk 4 above)
Comment (Alexmhaley) [reply src/handlers/syncUsers/dataProcessors/getAuthData.ts]: Removed

File: src/handlers/syncUsers/dataProcessors/getClassRosterData.ts (hunk 5 above)
Comment (Alexmhaley) [reply src/handlers/syncUsers/dataProcessors/getClassRosterData.ts]: removed

File: src/types/interfaces.d.ts (hunk 6 above)
Comment (Alexmhaley) [reply src/types/interfaces.d.ts]: Yeah you're right, will update the name

File: src/types/interfaces.d.ts (hunk 8 above)
Comment (Alexmhaley) [reply src/types/interfaces.d.ts]: Have updated

File: src/types/interfaces.d.ts (hunk 7 above)
Comment (Alexmhaley) [reply src/types/interfaces.d.ts]: Have updated

**Pull request merged**